pkg/cluster/config.go

```diff
@@ -0,0 +1,27 @@
+package cluster
+
+import (
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+type Config struct {
+	Cluster               *anywherev1.Cluster
+	VSphereDatacenter     *anywherev1.VSphereDatacenterConfig
```

Comment:
what about cloudstack?

Reply:
Those can be added later as the need arises
Or if everyone wants that, I can add it to the scope of this feature and implement it separately

My goal here was to introduce the new pattern and tooling that can be reused by everyone, not to refactor all existing providers